include/qpid/dispatch/iterator.h

```diff
@@ -329,6 +329,11 @@ int qd_iterator_ncopy(qd_iterator_t *iter, unsigned char* buffer, int n);
  */
 unsigned char *qd_iterator_copy(qd_iterator_t *iter);
 
+/**
```

Comment:
Is there any reason that the const version shouldn't be the only version?  i.e. add const to the argument of the original function?

Reply:
The original is documented as advancing the iterator past the data copied out.  I'm assuming that some codepath relies on that behavior. 

I'll need to audit the existing use cases before we can be sure we can safely make that change.

Reply:
In that case, don't make the change.